document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoProjectionSelectingExpressionParser.java

```diff
@@ -0,0 +1,95 @@
+package org.hypertrace.core.documentstore.mongo.parser;
+
+import static org.hypertrace.core.documentstore.mongo.MongoUtils.PREFIX;
+
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.core.documentstore.expression.impl.AggregateExpression;
+import org.hypertrace.core.documentstore.expression.impl.ConstantExpression;
+import org.hypertrace.core.documentstore.expression.impl.FunctionExpression;
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+import org.hypertrace.core.documentstore.expression.type.SelectingExpression;
+
+@Slf4j
+public final class MongoProjectionSelectingExpressionParser extends MongoSelectingExpressionParser {
+  private final String alias;
+
+  MongoProjectionSelectingExpressionParser(
+      final String alias, final MongoSelectingExpressionParser baseParser) {
+    super(baseParser);
+    this.alias = alias;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public Map<String, Object> visit(final AggregateExpression expression) {
+    assert baseParser != null;
```

Comment:
why would the `basePraser` be null? Is it possible to avoid that?

Reply:
Got rid with NoOpParser.